src/python/grpcio_observability/grpc_observability/BUILD.bazel

```diff
@@ -0,0 +1,73 @@
+# Copyright 2023 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("//bazel:cython_library.bzl", "pyx_library")
+
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+    name = "observability",
+    srcs = [
+        "client_call_tracer.cc",
+        "observability_main.cc",
+        "python_census_context.cc",
+        "sampler.cc",
+        "server_call_tracer.cc",
+    ],
+    hdrs = [
+        "client_call_tracer.h",
+        "constants.h",
+        "observability_main.h",
+        "python_census_context.h",
+        "sampler.h",
+        "server_call_tracer.h",
+    ],
+    includes = ["."],
+    deps = [
+        "//:grpc",
```

Comment:
When we get to the setuptools build, we're going to need to be very careful to ensure that the build only includes the code that is actually referenced from our o11y module otherwise the shared object library is going to be huge.